units/event_metajob.go

```diff
@@ -281,22 +286,48 @@ func (j *eventMetaJob) Run(ctx context.Context) {
 
 	j.AddError(j.dispatchUnprocessedNotifications(ctx))
 
-	j.events, err = event.FindUnprocessedEvents()
+	settings := j.env.Settings()
+	limit := settings.Notify.EventProcessingLimit
+	if limit <= 0 {
+		limit = evergreen.DefaultEventProcessingLimit
+	}
+	j.events, err = event.FindUnprocessedEvents(limit)
 	if err != nil {
 		j.AddError(err)
 		return
 	}
 
+	j.AddError(errors.Wrap(j.logEventCount(len(j.events), limit), "can't log unprocessed event count"))
+
 	if len(j.events) == 0 {
-		grip.Info(message.Fields{
-			"job_id":  j.ID(),
-			"job":     eventMetaJobName,
-			"time":    time.Now().String(),
-			"message": "no events need to be processed",
-			"source":  "events-processing",
-		})
 		return
 	}
 
 	j.AddError(j.dispatchLoop(ctx))
 }
+
+func (j *eventMetaJob) logEventCount(eventCount, limit int) error {
+	if eventCount == limit {
+		var err error
+		eventCount, err = event.CountUnprocessedEvents()
+		if err != nil {
+			grip.Error(message.Fields{
```

Comment:
I think this is redundant, since amboy will log errors added to a job.

Reply:
alas, we swallow errors when creating jobs, because of the options we pass to the queue population operations.

Reply:
when creating jobs, sure, but this is returned to a thing that calls `j.AddError`

Reply:
removed